contracts/earn/strategies/Alpaca/StrategyAlpaca.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import '@openzeppelin/contracts/math/SafeMath.sol';
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/utils/Pausable.sol';
+import '@uniswap/lib/contracts/libraries/TransferHelper.sol';
+
+import '../../../interfaces/Alpaca/IAlpacaVault.sol';
+import '../../../interfaces/Alpaca/IFairLaunch.sol';
+import '../../../interfaces/PancakeSwap/IPancakeRouter.sol';
+import '../../../interfaces/IWooAccessManager.sol';
+import '../../../interfaces/IStrategy.sol';
+import '../../../interfaces/IWETH.sol';
+import "../BaseStrategy.sol";
+
+contract StrategyAlpaca is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /* ----- State Variables ----- */
+
+    address public alpacaVault;
+    address public fairLaunch;
+    uint256 public immutable pid;
+
+    address[] public rewardToWantRoute;
+
+    /* ----- Constant Variables ----- */
+
+    address public constant wrappedEther = address(0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c);
+    address public constant reward = address(0x8F0528cE5eF7B51152A59745bEfDD91D97091d2F);
+    address public constant uniRouter = address(0x10ED43C718714eb63d5aA57B78B54704E256024E);
+
+    constructor(
+        address initVault,
+        address initAccessManager,
+        address initWant,
+        address initAlpacaVault,
+        address initFairLaunch,
+        uint256 initPid,
+        address[] memory initRewardToWantRoute
+    ) public BaseStrategy(initVault, initAccessManager) {
+        want = initWant;
+        alpacaVault = initAlpacaVault;
+        fairLaunch = initFairLaunch;
+        pid = initPid;
+        rewardToWantRoute = initRewardToWantRoute;
+
+        _giveAllowances();
+    }
+
+    /* ----- Public Functions ----- */
+
+    function harvest() public override whenNotPaused {
+        require(msg.sender == tx.origin || msg.sender == address(vault), 'StrategyAlpaca: EOA_or_vault');
+
+        IFairLaunch(fairLaunch).harvest(pid);
+        uint256 rewardBalance = IERC20(reward).balanceOf(address(this));
+
+        if (rewardBalance > 0) {
+            uint256 wantBalBefore = IERC20(want).balanceOf(address(this));
+            _swapRewards();
+            uint256 wantBalAfter = IERC20(want).balanceOf(address(this));
+            uint256 perfAmount = wantBalAfter.sub(wantBalBefore);
+            chargePerformanceFee(perfAmount);
+            deposit();
+        }
+    }
+
+    function deposit() public override whenNotPaused {
```

Comment:
Need to add this check, right?

   require(msg.sender == tx.origin || msg.sender == address(vault), 'StrategyAlpaca: EOA_or_vault');

Reply:
don't need this check